src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
+        while (currentNode != null) {
+            if (object == currentNode.getElement()
+                    || (object != null && object.equals(currentNode.getElement()))) {
+                removeNode(currentNode);
+                return true;
+            }
+            currentNode = currentNode.getNext();
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index, int range) {
+        if (index < 0 || index > range) {
+            throw new IndexOutOfBoundsException("Index [" + index
+                    + "] is out of available range. Available range is [0 - " + range + "]");
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> node;
+        int nodeIndex;
+        if (index > size >> 1) {
+            node = lastNode;
+            nodeIndex = size - 1;
+            while (nodeIndex > index) {
+                node = node.getPrev();
+                nodeIndex--;
+            }
+            return node;
+        }
+        node = firstNode;
+        nodeIndex = 0;
+        while (nodeIndex < index) {
+            node = node.getNext();
+            nodeIndex++;
+        }
+        return node;
+    }
+
+    private void putNodeMiddle(T value, Node<T> operationNode) {
+        Node<T> newNode = new Node<>(value, operationNode.getPrev(), operationNode);
+        operationNode.getPrev().setNext(newNode);
+        operationNode.setPrev(newNode);
+        size++;
+    }
+
+    private void putNode(T value, String position) {
+        Node<T> newNode;
+        switch (position) {
+            case "top":
+                newNode = new Node<>(value, lastNode, null);
+                lastNode.setNext(newNode);
+                lastNode = newNode;
+                break;
+            case "bottom":
+                newNode = new Node<>(value, null, firstNode);
+                firstNode.setPrev(newNode);
+                firstNode = newNode;
+                break;
+            default:
+                newNode = new Node<>(value, null, null);
+                firstNode = newNode;
+                lastNode = newNode;
+        }
+        size++;
+    }
+
+    private void removeNode(Node<T> node) {
+        if (!node.equals(firstNode) && !node.equals(lastNode)) {
+            node.getPrev().setNext(node.getNext());
+            node.getNext().setPrev(node.getPrev());
+        }
+        if (node.equals(firstNode) && size > 1) {
+            node.getNext().setPrev(null);
+            firstNode = node.getNext();
+        }
+        if (node.equals(lastNode) && size > 1) {
+            node.getPrev().setNext(null);
+            lastNode = node.getPrev();
+        }
+        if (firstNode.equals(lastNode)) {
+            firstNode = null;
+            lastNode = null;
+        }
+        size--;
```

Comment:
use == to check if node **is the same** (reference is the same) object or is equal